packages/react/src/analytics/integrations/GA4/GA4.ts

```diff
@@ -399,7 +408,18 @@ class GA4 extends integrations.Integration<GA4IntegrationOptions> {
         `context.event.${utils.ANALYTICS_UNIQUE_EVENT_ID}`,
       ) as string;
 
+      const coreVersion = get(
+        data,
+        `context.event.${utils.ANALYTICS_PACKAGE_VERSION}`,
+        '',
+      ) as string;
+
+      const packageVersion = `${PACKAGE_NAME}@${PACKAGE_VERSION};`;
```

Comment:
This can be calculated outside of the track method. In fact, you could add this calculation in the analytics facade for the react and pass it in the event context just the way that the Analytics class in analytics package does. That way you can share that value with all integrations that need it.